src/briefcase/console.py

```diff
@@ -41,6 +47,33 @@ class RichConsoleHighlighter(RegexHighlighter):
 
 rich_console = RichConsole(highlighter=RichConsoleHighlighter(), emoji=False)
 
+# Create console that records all logging (but prints nothing to screen) to save to file later.
+LOG_FILE_WIDTH = 150
+rich_log_file_console = RichConsole(
+    file=open(os.devnull, "w", encoding="utf-8", errors="ignore"),
```

Comment:
This is where I landed for a couple reasons:

When you set `record=True`, Rich will only save the content being logged if it is actually logged; therefore, the content has to actually be "written" to whatever medium you've configured Rich to write to. I initially noticed Rich supported a "quiet" mode where output is disabled; unfortunately, that applies to everything including recording. So, for `record=True` to be useful, the content _must_ be "written" somewhere.

Because the content must be written, just using the original `rich_console` wouldn't be workable....since, for instance, it wouldn't be possible to record `debug` output unless you're actually showing debug output on the screen. That led to creating this new `Console` altogether.

Additionally, the Rich consoles need to exist quite soon after briefcase starts running....but that creates a chicken and egg problem with knowing the exact filepath you'd want to ultimately save the log to.

All this culminated to creating a Console that writes to nowhere but records everything (whether its on the screen or not) thereby making all this data handily available to save off at the end....or mindlessly discard at exit.

I've tried to capture the gist of this in `console.py`.